frontend/public/components/row-filter.jsx

```diff
@@ -71,11 +69,34 @@ class CheckBoxes_ extends React.Component {
       } catch (ignored) {
         // ignore
       }
-
       this.setState({selected}, () => this.applyFilter());
     }
   }
 
+  toggle(event, itemId) {
+    event.preventDefault();
+    const selected = _.xor(this.state.selected, [itemId]);
+    this.setQueryParameters(selected);
+  }
+
+  selectAll() {
+    const selected = _.map(this.props.items, 'id');
+    this.setQueryParameters(selected);
+  }
+
+  countNumItems() {
+    const selectedCheckboxes = _.map(this.state.selected, (id) => {
```

Comment:
nit: I think `selectedCheckboxCounts` is more clear since it's an array of numbers, not checkboxes

```suggestion
    const selectedCheckboxCounts = _.map(this.state.selected, (id) => {
```